### PR TITLE
Block prefix ambiguity

### PR DESCRIPTION
**Pull request by jonny-no1**
For the reasoning behind this note please see issue https://github.com/symfony/symfony/issues/18953.

I based my branch on `2.7` as this is when the issue first arose.

### CODE REVIEW

File: book/forms.rst [hunk 1]
```diff
@@ -1122,6 +1122,23 @@ the choice is ultimately up to you.
             ));
         }
 
+.. caution::
+
+    When the name of your form class matches any of the built-in field types,
+    your form might not be rendered correctly. A form
```
Comment (javiereguiluz) [inline book/forms.rst]: `... might not be rendered ...` -> `... won't be rendered ...`

File: book/forms.rst [hunk 2]
```diff
@@ -1122,6 +1122,23 @@ the choice is ultimately up to you.
             ));
         }
 
+.. caution::
+
+    When the name of your form class matches any of the built-in field types,
```
Comment (xabbuh) [inline book/forms.rst]: This does not only apply to the built-in types, but is also true for any other form type that is coming from third-party bundles/libraries.

**Comment by foaly-nr1** [comment]
@javiereguiluz @xabbuh all done.

**Comment by stof** [comment]
@foaly-nr1 when you base your branch on 2.7, you should also select 2.7 as the target branch for the PR. Otherwise, the diff gets weird and the review is harder

**Comment by foaly-nr1** [comment]
Sorry my bad. Closing this to target `symfony:2.7`

**Pull request closed without merging**